components/stepper/index.css

```diff
@@ -170,29 +170,28 @@
 		--mod-textfield-border-color-disabled: var(--spectrum-stepper-border-color-disabled);
 	}
 
-	&:hover:not(.is-invalid, .is-disabled, .is-focused) {
+	&:hover:not(.is-invalid, .is-disabled, .is-focused, .is-keyboardFocused) {
 		--mod-infield-button-border-color: var(--mod-stepper-buttons-border-color-hover, var(--spectrum-stepper-buttons-border-color-hover));
 	}
 
 	&:not(.is-disabled) {
-		.is-focused,
+		&.is-focused,
 		&:focus {
 			--mod-infield-button-border-color: var(--highcontrast-stepper-border-color-focus, var(--mod-stepper-buttons-border-color-focus, var(--spectrum-stepper-buttons-border-color-focus)));
-			--mod-textfield-focus-indicator-width: 0;
+			--mod-textfield-focus-indicator-width: var(--mod-stepper-focus-indicator-width, 0);
 
 			&:hover {
-				/* stylelint-disable-next-line spectrum-tools/no-unknown-custom-properties -- allows for --spectrum-stepper-buttons-border-color-focus-hover to be defined outside of current context */
-				--mod-infield-button-border-color: var(--mod-stepper-buttons-border-color-focus-hover, var(--spectrum-stepper-buttons-border-color-focus-hover));
-				--mod-textfield-focus-indicator-width: 0;
-				--mod-textfield-border-color: var(--highcontrast-stepper-border-color-focus-hover, var(--mod-stepper-border-color-focus-hover, var(--spectrum-stepper-border-color-focus-hover)));
+				--mod-infield-button-border-color: var(--mod-stepper-buttons-border-color-focus-hover, var(--spectrum-stepper-border-color-focus-hover));
```

Comment:
The simplest solution to this border color issue was using --spectrum-stepper-border-color-focus-hover, instead of `--spectrum-stepper-buttons-border-color-focus-hover`. That kept coming up undefined, even though it was in the `spectrum-two.css` file.